hydra-cluster/src/Hydra/Generator.hs

```diff
@@ -130,3 +125,59 @@ genDatasetConstantUTxO faucetSk nClients nTxs = do
     pure ClientDataset{clientKeys, initialUTxO, txSequence}
 
   thrd (_, _, c) = c
+
+-- Prepare funding transaction which will give every client's
+-- 'externalSigningKey' "some" lovelace. The internal 'signingKey' will get
+-- funded in the beginning of the benchmark run.
+makeGenesisFundingTx :: SigningKey PaymentKey -> [ClientKeys] -> Gen Tx
+makeGenesisFundingTx faucetSk clientKeys = do
+  let nClients = length clientKeys
+  clientFunds <- forM clientKeys $ \ClientKeys{externalSigningKey} -> do
+    amount <- Coin <$> choose (1, availableInitialFunds `div` fromIntegral nClients)
+    -- NOTE: It seems if this becomes choose (1,1) we get funny signingKeys;
+    -- i.e. like "0001010100010001000000010100000001010001000101000000010101010001".
+    pure (getVerificationKey externalSigningKey, amount)
+  let fundingTransaction =
+        mkGenesisTx
+          networkId
+          faucetSk
+          (Coin availableInitialFunds)
+          clientFunds
+  pure fundingTransaction
+
+genDatasetConstantUTxODemo ::
+  -- | The faucet keys
+  (VerificationKey PaymentKey, SigningKey PaymentKey) ->
+  -- | Clients
+  [ClientKeys] ->
+  -- | Number of transactions
+  Int ->
+  NetworkId ->
+  SocketPath ->
+  IO Dataset
+genDatasetConstantUTxODemo (faucetVk, faucetSk) allClientKeys nTxs networkId' nodeSocket = do
```

Comment:
Have you tried this function against the freshly seeded Head? I think it would just work too and might be deterministic .. so maybe no need for `mkGenesisTx`. Alternatively, doing the querying outside and passing in a `UTxO` owned by the faucet to be spent should allow us to re-use more code.

Should: not duplicate the funding transaction logic